File: Benchmarks/normal/java/spring/testcode/BenchmarkTest14022.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest14022 {

    @PostMapping(path="/BenchmarkTest14022", consumes="application/xml")
    public void BenchmarkTest14022(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = String.format("payload=%s", xmlValue);
        if ("https://app.acmecdn.net".equals(data)) response.setHeader("Access-Control-Allow-Origin", data);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}

File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_14022.py
# SPDX-License-Identifier: Apache-2.0
from flask import request
import os


def BenchmarkTest14022():
    referer_value = request.headers.get('Referer', '')
    data = str(referer_value).replace('\x00', '')
    with open(os.path.join('/var/app/data', str(data)), 'r') as fh:
        content = fh.read()
    return content
